### PR TITLE
feat: add DFT loss and refactor SFT losses into submodule

### PR DESCRIPTION
**Pull request by kryvokhyzha**
# PR: Add Dynamic Fine-Tuning (DFT) loss and refactor SFT losses into submodule

## Description

This PR introduces **Dynamic Fine-Tuning (DFT)** loss support.

### Refactor: `tunix/sft/losses/` submodule

The existing `_default_loss_fn` is moved into a proper `losses/` package with shared building blocks:

- `per_token_cross_entropy(model, ...)` - runs the forward pass and returns raw per-token cross-entropy values with the target mask.
- `aggregate_loss(per_token_loss, mask)` - masks and averages per-token losses over valid tokens.
- `cross_entropy_loss_fn(model, ...)` - standard next-token prediction NLL (drop-in replacement for the old `_default_loss_fn`).

### New: DFT loss

Based on the paper [*On the Generalization of Supervised Fine-Tuning: A Reinforcement Learning Perspective with Reward Rectification*](https://arxiv.org/abs/2508.05629), DFT rescales the per-token cross-entropy loss by the model's own predicted probability (with stop-gradient):

```
dft_loss_t = stop_gradient(p(y_t)) * (-log p(y_t))
```

The implementation composes on top of the standard cross-entropy building blocks:

- `dft_rescale(per_token_loss)` - applies the per-token rescaling, usable with any per-token loss computation.
- `dft_loss_fn(model, ...)` - drop-in replacement for `cross_entropy_loss_fn`:

**Reference**

- Paper: [On the Generalization of Supervised Fine-Tuning: A Reinforcement Learning Perspective with Reward Rectification](https://arxiv.org/abs/2508.05629)

**Checklist**

- [X] I have added all the necessary unit tests for my change.
- [X] I have verified that my change does not break existing code and all unit tests pass.
- [X] I have added all appropriate doc-strings/documentation.
- [X] My PR is based on the latest changes of the main branch (if unsure, rebase the code).
- [X] I have signed the [Contributor License Agreement](https://cla.developers.google.com/about).
- [ ] I have followed [Contribution Guidelines](https://github.com/google/tunix/blob/main/docs/contributing.md).

### CODE REVIEW

File: tunix/sft/losses/cross_entropy.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  # Exclude the last step as it does not appear in the targets.
+  logits = logits[:, :-1, :]
+  target_tokens = input_tokens[:, 1:]
+  target_mask = input_mask[:, 1:]
+
+  # Convert the target labels to one-hot encoded vectors.
+  one_hot = jax.nn.one_hot(target_tokens, logits.shape[-1])
+
+  # Per-token cross-entropy: -log p(y_t).
+  xent = -jnp.sum(jax.nn.log_softmax(logits) * one_hot, axis=-1)
```
[the remark this answers is not in the document]
Comment (kryvokhyzha) [reply tunix/sft/losses/cross_entropy.py]: Done
